### ISSUE TITLE
How to use pretrained models.

### ISSUE THREAD

**Issue by KhurramPirov** [deepinsight/insightface#800]
I try to load pretrained model but get zero accuracy in the begining. In train.py I am doing the following way:

` sym, arg_params, aux_params = mx.model.load_checkpoint(os.path.join(args.pretrained, "model"), args.pretrained_epoch)
    sym = get_symbol(args)
    model = mx.mod.Module(
        context=ctx,
        symbol=sym,
    )
    model.bind(data_shapes=train_dataiter.provide_data, label_shapes=train_dataiter.provide_label)
    model.set_params(arg_params, aux_params)

    model.fit(train_dataiter,
        num_epoch=999999,
        eval_data=val_dataiter,
        eval_metric=eval_metrics,
        kvstore=args.kvstore,
        optimizer=opt,
        initializer=initializer,
        allow_missing=True,
        batch_end_callback=_batch_callback,
        arg_params=arg_params,
        aux_params=aux_params)`

**Comment by a26908456** [comment]
Hi, I also encountered the same problem. And I find solution here #569 .

Check `config.ckpt_embedding` in your config.py, it's True as default. 
If `config.ckpt_embedding` is set to True, the checkpoint only save the model until `embedding` layer. The other weights are **lost** such as `fc7_weight`. This led to the low accuracy when fine-tuning.

The conclusion is, if I want to train A model from scratch and take A as a pretrained model for training B model, `config.ckpt_embedding` should be set to False while training A and be set to True while training B. (If B model is final model I need.)

**Comment by vkasojhaa** [comment]
@a26908456 @nttstar I am little confused here. Suppose I am trying to fine tune the pretrained model provided [here](https://github.com/deepinsight/insightface/wiki/Model-Zoo) on my own dataset. Should I set `config.ckpt_embedding` as True or False. I want to use my fine tuned model for prediction. 
I see [here](https://github.com/deepinsight/insightface/issues/569) that @nttstar said to set `config.ckpt_embedding` as False while training your pre-trained model, so that fc7 weight matrix can be saved.
Note: I have tried to fine tune by setting it as False and getting good training accuracy on my dataset starting from epoch 1 itself. Can I use this model for prediction? Also I wanted to know when is the `fc7 weight` used?

**Comment by a26908456** [comment]
Hi @vkasojhaa,
Generally, we take the output of embedding layer as feature of an image. Then calculate cosine similarity between two features as their similarity. 
`fc7_weight` is only used while training and validating. It is used for classifying an input image to a certain class. If the person is not in your training set, it impossible to get correct result. That's why we use cosine similarity.

**Comment by vkasojhaa** [comment]
@a26908456 Thanks for replying. So, if I fine tuned my model with `config.ckpt_embedding` as False (so that fc7 weights are saved) and tried to use this model to get features will there be any problem? I guess the embedding layer will be used to get feature of an image in this case also.

**Comment by a26908456** [comment]
Hi @vkasojhaa,
I think it's ok. But the default output of the model with fc7 layer is classification results. I guess it is necessary to get internal layer after load checkpoint. 
see Extract Features in https://github.com/dmlc/mxnet-notebooks/blob/master/python/how_to/predict.ipynb

**Comment by vkasojhaa** [comment]
@a26908456 Got it. Thanks!
